cap/modules/imageimporter/image_importer.py

```diff
@@ -0,0 +1,75 @@
+from subprocess import check_call, CalledProcessError
+import docker
+import tqdm
+
+
+class ImageImporter:
+    """ ImageImporter saves docker images
+
+    Attributes:
+        image_name  eg. gitlab-registry.cern.ch/atrisovi/test-dockint
+        image_tag   optional
+        location    location to save the image at
+        username    username
+        token       private token
+    """
+
+    def __init__(self, username=None, token=None):
+        self.auth_config = {
+            "username": username,
+            "password": token
+        }
+        self.username = username
+        self.token = token
+        self.image_name = None
+        self.image_tag = None
+        self.location = None
+
+    def archive_image(self, image_name, location, image_tag=None):
+        """ Archives image """
+        if ':' in image_name:
+            self.image_name, self.image_tag = image_name.split(':')
+        else:
+            self.image_name = image_name
+            self.image_tag = "latest"
+        if image_tag is not None:
+            self.image_tag = image_tag
+        if location[-1] == '/':
+            location = location[:-1]
+        self.location = '{}/{}_{}'.format(location, self.image_name.replace('/', '_'),
+                                          self.image_tag)
+
+        try:
+            self.archive_image_with_python()
+        except Exception, e:
+            try:
+                print('Handling error:', e)
+                print('Trying another method...')
+                self.archive_image_with_skopeo()
+            except CalledProcessError, err:
+                print('Something went wrong.')
+                print('Handling error:', err)
+                return
+        print "The image was successfully saved."
+
+    def archive_image_with_python(self):
+        """ Archives image with docker-py. """
+        client = docker.from_env()
+        img = client.images.pull(self.image_name, tag=self.image_tag,
+                                 auth_config=self.auth_config)
+        f = open('{}.tar'.format(self.location), 'w')
```

Comment:
Here instead of opening/closing the file you can do 
```
with open(...) as f:
     ....
```

Reply:
👍🏻